testcases/PowerNVDump.py

```diff
@@ -1324,6 +1330,262 @@ def runTest(self):
                 self.fail("KernelArgTest failed to update kernel args")
 
 
+class OpTestWatchdog(PowerNVDump):
+    '''
+    This test verifies "watchdog module" with diffrent scenarios like
+    1. Watchdog module load and unload.
+    2. watchdog with action 1/ Reboot LPAR.
+    3. watchdog with action 0/ Halt LPAR.
+    4. watchdog with action 2 /dump collect
+                a. dump collect to a local disk
+                b. dump collect to a NFS disk
+                c. dump collect to a SAN FC disk
+    '''
+
+    def setUp(self):
+        super(OpTestWatchdog, self).setUp()
+        self.mg_system = self.cv_HMC.mg_system
+
+        if not self.cv_HMC.is_lpar_in_managed_system(self.mg_system, self.cv_HMC.lpar_name):
+            raise OpTestError("Lpar %s not found in managed system %s" % (
+                              self.cv_HMC.lpar_name, self.mg_system))
+
+    def get_watchdog_tool(self):
+        '''
+        This funtions copies the watchdog ".c" file from
+        test_binaries to root directory and compiles and
+        creates "watchdog-countdown" tool directory,using
+        this can trigger the watchdog events.
+        '''
+        filename = "watchdog-countdown.c"
+        self.cv_HOST.copy_test_file_to_host(filename, dstdir="/root")
+        base_file = os.path.splitext(filename)[0]
+        if not self.cv_HOST.host_run_command(f"make {base_file}"):
+            self.fail("make file command failed,Please check the file")
+
+    def module_load_with_parameters(self, i_module, timeout, action):
+        '''
+        This function will load the module using modprobe
+        with timeout and action parameters and valiadates
+        module load.
+        :params
+            i_module: watchdog Module name "pseries_wdt".
+            timeout: timeout value while loading module
+            action: type of reset.
+                    Action_0 : halt the LPAR
+                    Action_1 : reboot the LPAR
+                    Action_2 : collect dump and reboot LPAR
+        :rtype int
+        '''
+        try:
+            cmd = f"modprobe {i_module} timeout={timeout} action={action}"
+            self.cv_HOST.host_run_command(cmd)
+            self.script_timeout = timeout
+            self.script_action = action
+            return self.script_timeout, self.script_action
+
+        except CommandFailed as c:
+            l_msg = "Error in loading the module %s, modprobe failed: %s" % (
+                i_module, str(c))
+            raise OpTestError(l_msg)
+
+    def validate_timeout_and_action(self):
+        '''
+        Funtion to validate timeout and action of watchdog module.
+        '''
+        self.get_watchdog_timeout_value()
+        self.get_watchdog_action_value()
+
+        if (self.system_timeout_value == self.script_timeout) and (self.watchdog_mode == self.script_action):
```

Comment:
self.script_timeout and self.script_action does not exists

Reply:
@abdhaleegit Please check line 1383 and 1384